core/conversion/conversion.cpp

```diff
@@ -337,12 +369,28 @@ void ConvertBlockToNetDef(
     } else if (to_eval) {
       auto eval = EvaluateNode(ctx, n);
       if (eval) {
-        if (!eval.value().isTensor()) {
+        if (n->outputs().size() > 1) { // For ListUnpack scenario
+          if (eval.value().isList()) {
```

Comment:
I feel like tuples feel like the better type to use here. I think functionally they operate the same but cognitively they are more in line with how people think about multiple returns from functions.